### PR TITLE
Mitigate CVE 2015-9284

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description

There's an old vulnerability that has been promoted to a CVE recently, which is blocking all builds. 
Basically omniauth links with GETs are in some cases subject to CSRF attacks.
Unfortunately there's no agreed upon solution at present. There's a gem that mitigates the problem but it conflicts with a [patch in our codebase](https://github.com/thepracticaldev/dev.to/blame/master/config/initializers/persistent_csrf_token_cookie.rb), in addition to the fact that I can't verify this gem is working correctly since we don't use Omniauth directly.

My proposed solution is this PR:

- convert all sign in links to POSTs
- ignore the CVE until a more permanent solution is found, I'll keep monitoring the issue in the foreseeable future

## Related Tickets & Documents

- https://github.com/omniauth/omniauth/pull/809 and discussion
- https://github.com/rubysec/ruby-advisory-db/pull/390 and discussion
- https://github.com/cookpad/omniauth-rails_csrf_protection

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: app/views/layouts/_signup_modal.html.erb [hunk 1]
```diff
@@ -17,12 +17,26 @@
...
+          data: { "no-instant": true },
+        ) do %>
       <img src="<%= asset_path("twitter-logo.svg") %>" class="icon-img" alt="Twitter logo" /> Auth With Twitter
-    </a>
-    <a href="/users/auth/github?state=signup-modal" class="sign-up-link cta" data-no-instant>
+    <% end %>
+
+    <%= link_to(
+          user_github_omniauth_authorize_path(state: "signup-modal"),
+          ),
```
Comment (maestromac) [inline app/views/layouts/_signup_modal.html.erb]: Syntax error here
```suggestion
```

Comment (rhymes) [reply app/views/layouts/_signup_modal.html.erb]: 😅

**Comment by benhalpern** [comment]
@rhymes I believe this won't cause any problems but I _believe_ the `POST` action won't actually work because we don't include the Rails `ujs` in the project.

When this app was first created, UJS depended on jQuery. So to get rid of all those bytes for performance, I removed it from the project, and it's used specifically when needed (the logout page included `application.js` in order to include it.

In Rails 5.1, UJS removed the jQuery dependency, but we never put it back in. I'm not sure its size, but it probably makes more sense now without the jQuery dependency.

I _believe_ we could merge this and, at worst, we are not yet actually making `POST` requests because it would fallback to just performing a `GET`.

**Comment by rhymes** [comment]
@benhalpern oooh, definitely didn't know UJS wasn't inside. I this this is safe because without it the links still go through as GETs. So basically we would be ignoring this vulnerability until there's a proper fix from omniauth

**Pull request merged**